consensus/pacemaker/pacemaker.go

```diff
@@ -0,0 +1,280 @@
+package pacemaker
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	consensusTelemetry "github.com/pokt-network/pocket/consensus/telemetry"
+	typesCons "github.com/pokt-network/pocket/consensus/types"
+	"github.com/pokt-network/pocket/runtime/configs"
+	"github.com/pokt-network/pocket/shared/codec"
+	"github.com/pokt-network/pocket/shared/modules"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+const (
+	DefaultLogPrefix    = "NODE"
+	pacemakerModuleName = "pacemaker"
+
+	// A buffer around the pacemaker timeout to avoid race condition; 30ms was arbitrarily chosen
+	timeoutBuffer = 30 * time.Millisecond
+
+	NewRound = typesCons.HotstuffStep_HOTSTUFF_STEP_NEWROUND
+	Propose  = typesCons.HotstuffMessageType_HOTSTUFF_MESSAGE_PROPOSE
+)
+
+type Pacemaker interface {
```

Comment:
Ditto regarding encorcing that a struct implements an interface: https://stackoverflow.com/questions/10498547/ensure-a-type-implements-an-interface-at-compile-time-in-go

```go
var _ Pacemaker = &paceMaker{}
```